apps/babelstream/package.py

```diff
@@ -0,0 +1,337 @@
+    version('main', branch='main')
+    maintainers = ["robj0nes ", "tomdeakin", "tom91136"]
+
+
+    #The CudaPackage at the top class gives us the following variants so no need to define them again
+    #https://github.com/spack/spack/blob/fce95e2efb76da3c6ee549f2a9b4e257fb42ae50/lib/spack/docs/build_systems/cudapackage.rst
+    # https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/cuda.py
+    # depends_on("cuda", when="+cuda")
+    #variant("cuda", default=False, description="Enable CUDA support")
+    #variant("cuda_arch", values=str, default="none", description="The CUDA architecture being used.")
+
+
+    variant("sycl", default=False, description="Enable SYCL support")
+    
+
+    #Defined in CudaPackage
+    #variant("cuda", default=False, description="Enable CUDA support")
+    variant("sycl2020", default=False, description="Enable SYCL support")
+    variant("omp", default=False, description="Enable OpenMP support")
+    variant("ocl", default=False, description="Enable OpenCL support") ## Crushes llvm requires RAM
+    variant("tbb", default=False, description="Enable TBB support")
+    variant("acc", default=False, description="Enable OpenACC support") 
+    variant("thrust", default=False, description="Enable THRUST support")
+    variant("raja", default=False, description="Enable RAJA support") 
+
+    variant("stddata", default=False, description="Enable STD-data support")
+    variant("stdindices", default=False, description="Enable STD-data support")
+    variant("stdranges", default=False, description="Enable STD-data support")
+    
+    variant("dir", values=str, default="none", description="Enable Directory support")
+
+    conflicts("dir=none",when="+kokkos",msg="KOKKKOS requires architecture to be specfied by dir=")
+    variant("kokkos", default=False, description="Enable KOKKOS support")
+    
+    #ACC conflict
+    variant("cpu_arch", values=str, default="none", description="Enable CPU Target for ACC")
+    # conflicts("cpu_arch=none",when="+acc",msg="KOKKKOS requires architecture to be specfied by dir=")
+    variant("target", values=str, default="none", description="Enable CPU Target for ACC")
+    
+    
+    # CUDA conflict 
+    # this is not provided in CudaPackage
+    conflicts("cuda_arch=none",when="+cuda",msg="CUDA requires architecture to be specfied by cuda_arch=")
+    variant("mem", values=str, default="DEFAULT", description="Enable MEM Target for CUDA")
+        
+    #download raja from https://github.com/LLNL/RAJA
+    conflicts("dir=none",when="+raja",msg="KOKKKOS requires architecture to be specfied by dir=")
+
+    #Thrust Conflict
+    conflicts("~cuda", when="+thrust", msg="Thrust requires +cuda variant")
+
+    #TBB Dependency
+    depends_on("intel-oneapi-tbb", when="+tbb")
+    
+    #Kokkos Dependency
+    depends_on("kokkos@3.7.1", when="+kokkos")
+
+    #OpenCL Dependency
+    depends_on("pocl@1.5", when="+ocl")
+
+
+
+    depends_on("cmake@3.14.0:3.23", type="build")
```

Comment:
No need to restrict future versions
```suggestion
    depends_on("cmake@3.14.0:", type="build")
```